src/server/pfs/s3/driver.go

```diff
@@ -54,47 +53,20 @@ func NewMasterDriver() *MasterDriver {
 	return &MasterDriver{}
 }
 
-func branchBucketName(b *pfs.Branch) string {
-	if b.Repo.Type == pfs.UserRepoType {
-		return fmt.Sprintf("%s.%s", b.Name, b.Repo.Name)
-	}
-	return fmt.Sprintf("%s.%s.%s", b.Name, b.Repo.Type, b.Repo.Name)
-}
-
-func projectBranchBucketName(b *pfs.Branch) string {
-	if b.Repo.Type == pfs.UserRepoType {
-		return fmt.Sprintf("%s.%s.%s", b.Name, b.Repo.Name, b.Repo.Project.Name)
-	}
-	return fmt.Sprintf("%s.%s.%s.%s", b.Name, b.Repo.Type, b.Repo.Name, b.Repo.Project.Name)
-}
-
 func (d *MasterDriver) listBuckets(pc *client.APIClient, r *http.Request, buckets *[]*s2.Bucket) error {
-	var isProjectAware = mux.Vars(r)[isProjectAwareVar] == isProjectAwareValue
-	repos, err := pc.ListRepoByType("") // get repos of all types
+	repos, err := pc.ListRepo() // get repos of all types
 	if err != nil {
 		return err
 	}
 
 	for _, repo := range repos {
-		if !isProjectAware && repo.Repo.Project.Name != pfs.DefaultProjectName {
-			continue // skip repos not in the default project
-		}
-		if repo.Repo.Type == pfs.SpecRepoType {
-			continue // hide spec repos, but allow meta/stats repos
-		}
 		t, err := types.TimestampFromProto(repo.Created)
 		if err != nil {
 			return err
 		}
-		for _, branch := range repo.Branches {
-			var name string
-			if isProjectAware {
-				name = projectBranchBucketName(branch)
-			} else {
-				name = branchBucketName(branch)
-			}
+		for _, b := range repo.Branches {
 			*buckets = append(*buckets, &s2.Bucket{
-				Name:         name,
+				Name:         fmt.Sprintf("%s.%s.%s", b.Name, b.Repo.Name, b.Repo.Project.Name),
```

Comment:
You might wish to either use `GetRepo().GetProject().GetName()` &c., or check for nils.

Reply:
Done!